file/readfile.go

```diff
@@ -87,6 +90,11 @@ func readContent(reader io.Reader) (*Content, error) {
 	if err != nil {
 		return nil, err
 	}
+	processedContent, err := processEnvs(string(bytes))
+	if err != nil {
+		return nil, fmt.Errorf("parsing file: %w", err)
+	}
+	bytes = []byte(processedContent)
```

Comment:
Please don't reuse the same `[]byte` for input and output of template rendering. From a strongly typed perspective, these two values represent different data types (unrendered template versus rendered document)

Reply:
I get the point somewhat but not enough.
Why does it really matter? Or maybe when?

I do agree that this way of thinking is correct, I just don't completely grasp it so please help me improve my understanding here.

And since I agree, I have already updated the patch,